index.js

```diff
@@ -1 +1,40 @@
-console.log('Happy developing ✨')
+const operationButtons = document.querySelectorAll('.operation-button');
+const equalButton = document.querySelector('#button-equal');
+const clearButton  = document.querySelector('#button-clear');
+const calculatorScreen = document.querySelector('#screen');
+const errorMessage = document.querySelector('.error-message');
+
+operationButtons.forEach(function(operationButton) {
+  operationButton.addEventListener('click', function() {
+    let operationValue = operationButton.innerText;
+
+    if (calculatorScreen.value.length < 9) {
+      calculatorScreen.value += operationValue;
+    } else {
+      errorMessage.innerText = 'max 9 digits';
+    }
+  })
+})
+
+equalButton.addEventListener('click', function() {
+  const screenValue = calculatorScreen.value;
+  const lastCharacterFromResult = screenValue.slice(-1);
+
+  if (lastCharacterFromResult.match(/[+-/*]/) || screenValue === '') {
+    errorMessage.innerText = 'wrong format';
+  } else {
+    let operationResult = new Function('return ' + screenValue)();
+    errorMessage.innerText = '';
+
+    if (screenValue.includes('.')) {
+      calculatorScreen.value = operationResult.toFixed(2);
```

Comment:
This is not the reason why 0.1 + 0.2 does not equal 0.3.
Do some research and come back with an answer why

![image](https://github.com/user-attachments/assets/1630315b-2427-48ed-ae82-b21895b5f7ea)

Reply:
i found a working workaround - please let me know if this solution is enough for this task

Reply:
Explain the workaround

Reply:
so basically i multiply the equation, round it and the devide by the same number that i multiply it to remove the unnecessary numbers

Reply:
Unfortunately, it does not seem to work correctly

![Screenshot from 2024-11-27 00-08-31](https://github.com/user-attachments/assets/0761e76b-5175-48e8-84aa-f7167496dbb1)
![Screenshot from 2024-11-27 00-08-35](https://github.com/user-attachments/assets/fc6830fb-6cdf-405c-acd4-82317cbd4355)